hydra-node/src/Hydra/Chain/Direct/Handlers.hs

```diff
@@ -109,19 +112,23 @@ mkChain tracer queryTimeHandle wallet headState submitTx =
               -- to bootstrap the init transaction. For now, we bear with it and
               -- keep the static keys in context.
               fromPostChainTx timeHandle wallet headState tx
-                >>= finalizeTx wallet headState . toLedgerTx
+                >>= finalizeTx tx wallet headState . toLedgerTx
             )
         submitTx vtx
+    , getUTxO = fromPairs . fmap toLedger . Map.assocs <$> atomically getUTxO
```

Comment:
This is not all "the `UTxO tx` it knows." (per haddock of this function). In fact, the `UTxO` of the tracked on-chain head state is missing and should be incorporated in this with something like `getKnownUTxO <$> readTVar headState` OR the name and documentation should be changed.

Reply:
Fair enough, happy to add the OCHS to the mix